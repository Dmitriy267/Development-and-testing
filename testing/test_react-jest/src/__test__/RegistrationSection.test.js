import React from 'react';
import {shallow} from 'enzyme';
import RegistrationSection from '../components/RegistrationSection';
describe ('RegistrationSection component',()=>{
	test ('should renders  witout errors',()=>{
		const component=shallow(<RegistrationSection/>);
	}),
	test('should renders correctly',()=>{
		const component=shallow(<RegistrationSection/>);
		expect (toJson(component)).toMatchSnapshot();
	}),
	test('check for the presence of the h1 tag',()=>{
		const component=shallow(<RegistrationSection/>);
		let tegComponent=component.find('h1');
		expect (tegComponent.length).toEqual(1);
	}),
	test('tag h1 has class',()=>{
		const component=shallow(<RegistrationSection/>);
		let tegComponent=component.find('h1');
		expect(tegComponent.hasClass('registration-section__title')).toEqual(true);
	})
})