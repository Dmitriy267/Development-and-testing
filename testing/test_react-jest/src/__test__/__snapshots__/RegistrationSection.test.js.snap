// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegistrationSection component should renders correctly 1`] = `
<section
  className="registration-section"
>
  <img
    alt="Рисунок главной страницы по каршерингу"
    className="registration-section__images_size"
    src="test-file-stub"
  />
  <h1
    className="registration-section__title"
  >
    Каршеринг в любой точке России
  </h1>
  <p
    className="registration-section__text"
  >
    Будьте всегда за рулём во время путешествий и командировок.
  </p>
  <button
    className="registration-btn"
  >
    Зарегистрироваться
  </button>
</section>
`;
